core/vm/evm.go

```diff
@@ -122,9 +124,14 @@ type BlockContext struct {
 	TransferMultiCoin TransferMCFunc
 	// GetHash returns the hash corresponding to n
 	GetHash GetHashFunc
-	// PredicateResults are the results of predicate verification available throughout the EVM's execution.
-	// PredicateResults may be nil if it is not encoded in the block's header.
+	// PredicateResults are the results of predicate verification available
+	// throughout the EVM's execution.
+	//
+	// PredicateResults may be nil if it is not encoded in the extra field of
+	// the block's header or if the extra field has not been parsed yet.
 	PredicateResults *predicate.Results
+	// Extra is the extra field from the block header.
+	Extra []byte
```

Comment:
(info) `libevm` makes the entire [`Header` available in the `BlockContext`](https://github.com/ava-labs/libevm/blob/d32c7e0090fc380d8b9c36cccd82fdc38e7ffe95/core/vm/evm.go#L83) in lieu of adding the `PredicateResults`. Given the comment on `Extra` here, this should therefore be compatible.